### PR TITLE
Adds functions emulating the C++ glog's CHECK macros.

### PR DESCRIPTION
**Pull request by heathkh**
The c++ glog includes lots of handy invariant checks that prints helpful messages.  I've recreated most of these in my own glog.py.  Here's a pull-request that adds these check functions, updates the docs, and adds a unittest.

### CODE REVIEW

File: glog.py [hunk 1]
```diff
@@ -102,7 +126,7 @@ def format(self, record):
             level = '?'
         date = time.localtime(record.created)
         date_usec = (record.created - int(record.created)) * 1e6
-        record_message = '%c%02d%02d %02d:%02d:%02d.%06d %s %s:%d] %s' % (
+        record_message = '[%c%02d%02d %02d:%02d:%02d.%06d %s %s:%d] %s' % (
```
Comment (benley) [inline glog.py]: Is the leading left bracket actually part of the c++ glog format?

**Comment by benley** [comment]
Thanks!  I will take a good look at this later tonight.

**Comment by benley** [comment]
Haven't forgotten about this - sorry for the delay

File: glog.py (hunk 1 above)
Comment (heathkh) [inline glog.py]: The opening bracket was a mistake... probably result of mindless bracket pairing when touching code when sleepy.  Fixed it after I sent you the PR.

File: README.rst [hunk 2]
```diff
@@ -76,9 +76,46 @@ parsing flags, like so:
...
+
+
 Happy logging!
 
 .. _twitter.common.log: https://github.com/twitter/commons/tree/master/src/python/twitter/common/log
 .. _never actually released: https://groups.google.com/d/msg/google-glog/a_JcyJ4p8MQ/Xu-vDPiuCCYJ
-.. _C++ glog module: https://github.com/google/glog
+.. _C++ glog library: https://github.com/google/glog
 .. _gflags: https://github.com/google/python-gflags
+.. _check macros: https://google-glog.googlecode.com/svn/trunk/doc/glog.html#check
```
Comment (benley) [inline README.rst:121]: This URL appears to be dead now.  Maybe substitute https://htmlpreview.github.io/?https://github.com/google/glog/master/doc/glog.html in its place?

File: glog.py [hunk 3]
```diff
@@ -169,3 +193,122 @@ def init():
 handler.setFormatter(GlogFormatter())
 logger.addHandler(handler)
 setLevel(FLAGS.verbosity)
+
+
+# Define functions emulating C++ glog check-macros 
+# https://google-glog.googlecode.com/svn/trunk/doc/glog.html#check
+
+def format_stacktrace(stack):
```
Comment (benley) [inline glog.py:201]: Is this stacktrace style derived from something in the c++ glog?

**Comment by benley** [comment]
Sorry it took me over three months to review this pull request!  I'm catching up on a lot of stuff now that I have time to do any of it.

**Pull request merged**